src/arraytraits.rs

```diff
@@ -84,7 +87,10 @@ impl<S, D, I> IndexMut<I> for ArrayBase<S, D>
     #[inline]
     fn index_mut(&mut self, index: I) -> &mut S::Elem {
         debug_bounds_check!(self, index);
-        self.get_mut(index).unwrap_or_else(|| array_out_of_bounds())
+        unsafe {
+            &mut *self.as_mut_ptr().offset(index.index_checked(&self.dim, &self.strides)
+                                                .unwrap_or_else(|| array_out_of_bounds()))
```

Comment:
For all the @bluss and other hot shots out there, you need to remember that the mutable case is really tricky with RcArray, which is clone-on-write. This code is wrong, because it uses the stride before calling as_mut_ptr.